micrometer-core/src/main/java/io/micrometer/core/annotation/Counted.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.annotation;
+
+import java.lang.annotation.*;
+
+/**
+ * Annotated methods would expose a few counter metrics about their execution status.
+ * By default, only failure attempts would be reported. Switching the {@link #successfulAttempts()}
+ * to {@code true} would instruct the corresponding aspect to record successful attempts, too.
+ *
+ * @author Ali Dehghani
+ * @see io.micrometer.core.aop.CountedAspect
+ */
+@Inherited
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Counted {
+
+    /**
+     * Represents the metric name for the to-be-recorded counters. The default value is
+     * {@code method.counted}.
+     *
+     * @return The metric name.
+     */
+    String value() default "";
+
+    /**
+     * By default, successful attempts won't be recorded. Switch it to {@code true} in order to
+     * record successful attempts alongside with failed attempts.
+     *
+     * @return Whether to report successful attempts or not.
+     */
+    boolean successfulAttempts() default false;
```

Comment:
I think defaulting this to true would be better. My comment on #1435 about defaulting to false was about the 'submitted' count (which you have eliminated)

Reply:
Does it make sense to make this behavior configurable?

Reply:
I could see that some users would want to `recordFailuresOnly`, if they only care about exceptions/failures. So I think there might be a good reason to have some configuration, but renaming it might be a good idea.

Reply:
I've renamed to `recordFailuresOnly`, as you suggested.